src/ITensorTDVP.jl

```diff
@@ -1,19 +1,15 @@
 module ITensorTDVP
-export TimeDependentSum, dmrg_x, linsolve, tdvp, to_vec
 
-include("defaults.jl")
-include("update_observer.jl")
+using Reexport: @reexport
```

Comment:
I don't think it is necessary to use `Reexport` here, since there aren't so many exports. The main motivation for that package is to automatically re-export a large number of exports that you don't want to write out by hand. Let's just explicitly write the exports as `export contract, dmrg_x, linsolve, ...`.

Reply:
In fact I don't see `contract` as being exported currently, so what you wrote introduces that as a new export which we want to avoid. The current export list is:
```julia
export TimeDependentSum, dmrg_x, linsolve, tdvp, to_vec
```
so we should just preserve that list of exports.

Additionally, we should test which types and functions are exported in the tests, like we do with `ITensors.jl`: https://github.com/ITensor/ITensors.jl/blob/v0.4.0/test/base/test_exports.jl. For example:
```julia
@eval module $(gensym())
using ITensorTDVP: ITensorTDVP
using Test: @test, @testset
@testset "Test exports" begin
  itensortdvp_exports = [
    :ITensorTDVP,
    :TimeDependentSum,
    :dmrg_x,
    :linsolve,
    :tdvp,
    :to_vec,
  ]
  @test issetequal(names(ITensorTDVP), itensortdvp_exports)
end
end
```
which also checks that we don't accidentally add more exports (`issetequal` will check if their lengths are equal).

Reply:
I added it in the latest update